Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -791,7 +802,10 @@ void ContainerNode::replaceAll(Node* node)
     Ref protectedThis { *this };
     ChildListMutationScope mutation(*this);
     NodeVector removedChildren;
-    auto didRemoveElements = removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren, DeferChildrenChanged::Yes);
+
+    auto [subTreeSize, didRemoveElements, canBeDelayed ] = removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren, DeferChildrenChanged::No);
+    if (this != &rootNode() && document().isAsyncNodeDeletionEnabled() && document().isOpportunisticTaskSchedulerEnabled() && canBeDelayed == ChildrenDeletionCanBeDelayed::Yes)
+        delayDeletingRemovedChildren(WTFMove(removedChildren), subTreeSize);
```

Comment:
This code will load `document()` thrice. Since `document()` involves two indirect loads, we should avoid that.